### DISCUSSION TITLE
How to have component (Image) be added in a new line above it's insertion?

### DISCUSSION THREAD

**Discussion by zacharykatsnelson** [ueberdosis/tiptap#2423] [General Questions]
Hello everyone,

On my bubble menu (at the very end) there is a button that adds an image using the [add image functionality](https://tiptap.dev/examples/images) as outlined in the documentation.

![example1](https://user-images.githubusercontent.com/61557157/150736168-c8455d1b-65ec-4411-ad5d-4fe2f527a2ac.jpg)

However, if I add an image from the bubble menu that highlights this text it will overwrite that text with the new image. The result will look like this:

![example2](https://user-images.githubusercontent.com/61557157/150736523-28ed9f76-11f2-49f5-80ba-72488eb4651f.jpg)

How can I make it so that the image of the penguin is inserted  directly **_above_** the highlighted node rather than overwriting it?

Any and all help is greatly appreciated.

**Comment by philippkuehn** [comment]
The `setImage` command is using the `insertContent` command for inserting the image. By default this command replaces the content of your selection.

https://github.com/ueberdosis/tiptap/blob/31efb0802e77e32c8c1b65e1a8456a5d3262b0ae/packages/extension-image/src/image.ts#L73-L78

You have to write a custom command and use the `insertContentAt` command to insert the image at a specific position based on your selection.

**Reply by zacharykatsnelson** [reply]
Thanks for the pointer @philippkuehn !

Here is my codeSandbox: https://codesandbox.io/s/busy-frog-8yexu?file=/src/App.tsx

So far I have been able to do something like 
```
const addImage = (props: any) => {
    const url = window.prompt('URL');
    console.log(editor.state.selection);
    if (url) {
      editor.chain().focus().insertContentAt(0, { type: 'image' }).setImage({ src: url }).run();
    }
  };
```
Which allows me to insert the image at the **very beginning** of the document but I have been unable to find a way to access the **node position** to be able to subtract from it.

Ideally I would like to do something like this:
```
const addImage = (props: any) => {
    const startPos = props.getPos() - props.node.nodeSize
    const url = window.prompt('URL');

    if (url) {
      editor
             .chain()
             .focus()
             .insertContentAt(startPos, {type: "image"})
             .run()
    }
  };
```
However I cannot find  a way to access the node position ( `props.node` is `undefined` and no way to `getPos()`). :thinking: Is there anything I am missing here?

**Comment by ricotrevisan** [comment]
Bumping this hoping you found a way to easily select where you want to insert the image.
